src/render_helpers/border.rs

```diff
@@ -43,6 +44,7 @@ impl BorderRenderElement {
     pub fn new(
         size: Size<f64, Logical>,
         gradient_area: Rectangle<f64, Logical>,
+        gradient_format: f32,
```

Comment:
Pass the enum here, and convert right before passing into the uniform. Render elements know what shaders they use and their input format; whoever uses the render elements doesn't.